### PR TITLE
Use locally served resources

### PR DESCRIPTION
**Pull request by ajarmoniuk**
Use locally served resources where possible in integration tests.

Some integration tests were failing because external resources they referred to were no longer available. That triggered this PR.

In quite a few cases, a unit test using Wiremock is much more suitable than an integration test involving maven-invoker-plugin.

Tests involving the creation of a "stamp" file in Groovy were failing on my machine because the plugin said it wasn't supported anymore.

**Ready to merge.**

### CODE REVIEW

**Comment by ajarmoniuk** [comment]
@longtimeago please merge

File: src/test/java/com/googlecode/download/maven/plugin/internal/WGetTest.java [hunk 1]
```diff
@@ -295,24 +322,306 @@ public void testCacheRetainingValuesFromTwoConcurrentCalls() throws Exception {
...
-        return clientBuilder;
+    private static String printStackTrace(Throwable t) {
+        return Arrays.stream(t.getStackTrace())
+                .map(StackTraceElement::toString)
+                .map(s -> "\t" + s)
+                .collect(Collectors.joining("\n"));
+    }
+
+    /**
+     * Testing {@code alwaysVerifySignature} with an already existing file. The plugin, if provided the {@code md5},
```
Comment (longtimeago) [inline src/test/java/com/googlecode/download/maven/plugin/internal/WGetTest.java:551]:
```suggestion
     * Testing {@code alwaysVerifyChecksum} with an already existing file. The plugin, if provided the {@code md5},
```

File: src/test/java/com/googlecode/download/maven/plugin/internal/WGetTest.java [hunk 2]
```diff
@@ -295,24 +322,306 @@ public void testCacheRetainingValuesFromTwoConcurrentCalls() throws Exception {
...
+            } catch (MojoExecutionException | MojoFailureException | IOException ex) {
+                final Throwable rootCause = getRootCause(ex);
+                fail("Execution failed with " + key + " test: " + rootCause + " stack trace:\n"
+                        + printStackTrace(rootCause));
+            }
+        });
+    }
+
+    /**
+     * Testing {@code alwaysVerifySignature} with an already existing file. The plugin, if provided the {@code md5},
```
Comment (longtimeago) [inline src/test/java/com/googlecode/download/maven/plugin/internal/WGetTest.java:586]:
```suggestion
     * Testing {@code alwaysVerifyChecksum} with an already existing file. The plugin, if provided the {@code md5},
```

**Review by longtimeago — APPROVED** [review]
@ajarmoniuk wesome contribution, thanks a lot! :+1:

**Pull request merged**